src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -470,7 +470,7 @@ trait TypeAssigner {
     tree.withType(sym.nonMemberTermRef)
 
   def assignType(tree: untpd.Annotated, annot: Tree, arg: Tree)(implicit ctx: Context) =
-    tree.withType(AnnotatedType(arg.tpe, Annotation(annot)))
+    tree.withType(AnnotatedType(arg.tpe.widen, Annotation(annot)))
```

Comment:
I'm not sure about this change, but without it there will be several TreeChecker errors. Could you please advise @smarter ? Thanks.